adr/ADR-13.md

```diff
@@ -0,0 +1,291 @@
+# Pull Subscribe internals
+
+| Metadata | Value                     |
+|----------|---------------------------|
+| Date     | 2021-07-20            |
+| Author   | wallyqs                   |
+| Status   | `Partially Implemented`   |
+| Tags     | jetstream, client         |
+
+## Motivation
+
+One of the forms of message delivery in JetStream is a pull based
+consumers. In this ADR it is described what is the current state of
+implementing consumption of messages for this type of consumers.
+
+## Context
+
+A pull based consumer is a type of consumer that does not have a
+`delivery subject`, that is the server _does not know_ where to send
+the messages.  Instead, the clients have to request for the messages
+to be delivered as needed from the server.  For example, given a stream `bar`
+with the consumer with a durable name `dur` (all pull subscribers have to be
+durable), a pull request would look like this in terms of the
+protocol:
+
+```shell
+PUB $JS.API.CONSUMER.MSG.NEXT.bar.dur _INBOX.x7tkDPDLCOEknrfB4RH1V7.UBZe2D 0
```

Comment:
I'd avoid showing this basic form, generally we should never make pulls that doesnt have `no_wait` or `expires` since they cause confusing behavior and pile ups on the server.  The design you outline does the right thing - but someone might see this example and think its a recommendation or something

Reply:
sounds good, updated to an example with `expires` instead